### ISSUE TITLE
[Mobile A11y] Expand/collapse buttons on Footer navigation menu are <h4> s but look and function like buttons

### ISSUE THREAD

**Issue by trimspa** [uswds/uswds#4475]
### **Describe the bug**
When accessing the Footer component on a mobile device, the navigation menu turns into a stack of 4 expand/collapse buttons indicating menu headers. These elements look and function like buttons (they can be expanded to reach the sub navigation) but are a series of h4s. A screen reader user would not be aware that these headings were actionable. The elements are also missing aria-expanded attributes. A screen reader users would not know whether the element was expanded or collapsed. 

**A note on keyboard accessibility:** menu buttons are navigable on an iPhone using VoiceOver and can be physically expanded/collapsed as expected. On a laptop using Responsive Device Mode on Chrome and Firefox Dev Tools, these elements were not accessible via keyboard. 

### **Steps to reproduce the bug**
1. Go to the [Footer Component](https://designsystem.digital.gov/components/footer/) on a mobile device. 
2. Turn on your devices' screen reader. 
3. Navigate to the first menu header (labeled here as "Primary Link").
3. Screen reader will indicate that the element is an h4.
4. Screen reader will not indicate that the element is clickable nor whether it is expanded or collapsed.

### **Expected behavior**
I would expect to hear a screen reader to report the elements' name (Primary Link), it's role (button) and whether the element is expanded or collapsed. 

### **Screenshots**

<img width="1218" alt="Screen Shot 2022-01-20 at 4 47 28 PM" src="https://user-images.githubusercontent.com/64163603/150434379-6aa2f7d6-f2ec-4e08-b467-ea150fbaec13.png">

**Comment by mahoneycm** [comment]
We should also address if these `Primary link` H4's should behave as links for the `Footer: big` component in desktop view. Currently, turns the mouse into a pointer as if it should be clicked.

![image](https://user-images.githubusercontent.com/25211650/150602232-0bf9ea95-adb4-4b22-a491-248537921f05.png)

**Comment by trimspa** [comment]
I think it would also be prudent to change the text here to "Heading" instead of "Primary link." Screen reader-wise, it's confusing to hear "Heading level 4, primary link." I realize this is an example of the component (and "Primary link" would be swapped out for a relevant heading, etc) but just to be super clear.
